Modules/TPC/src/TrendingTaskTPC.cxx

```diff
@@ -187,81 +260,166 @@ void TrendingTaskTPC::drawCanvas(TCanvas* thisCanvas, const std::string& var,
+    thisCanvas->cd(1);
 
+    myReader.SetEntry(nEntries - 1); // set event to last entry with index nEntries-1
+
+    int iEntry = 0;
     for (int p = 0; p < NuPa; p++) {
-      DataStorage[p][iEntry] = (DataRetrieveVector->at(p)).RetrieveValue(typeName);
 
+      double DataPoint = (DataRetrieveVector->at(p)).RetrieveValue(typeName);
+      double ErrorX = 0.;
+      double ErrorY = 0.;
       if (!err.empty()) {
-        ErrorX[p][iEntry] = (DataRetrieveVector->at(p)).RetrieveValue(errXName);
-        ErrorY[p][iEntry] = (DataRetrieveVector->at(p)).RetrieveValue(errYName);
-      } else {
-        ErrorX[p][iEntry] = 0.;
-        ErrorY[p][iEntry] = 0.;
+        ErrorX = (DataRetrieveVector->at(p)).RetrieveValue(errXName);
+        ErrorY = (DataRetrieveVector->at(p)).RetrieveValue(errYName);
       }
+      double xLabel = (DataRetrieveVector->at(p)).RetrieveValue("sliceLabelX");
+
+      graphErrors->SetPoint(iEntry, xLabel, DataPoint);
+      graphErrors->SetPointError(iEntry, ErrorX, ErrorY); // Add Error to the last added point
+
+      iEntry++;
     }
-    iEntry++;
-  }
 
-  // Fill the graph(errors) to be published.
-  for (int p = 0; p < NuPa; p++) {
-    thisCanvas->cd(p + 1);
-    graphPad = new TGraph(nEntries, TimeStorage, DataStorage[p]);
-    graphPad->Draw(opt.data());
+    if (myReader.Next()) {
+      ILOG(Error, Devel) << "Entry beyond expected last entry" << ENDM;
+    }
+
+    myReader.Restart();
 
-    // Draw errors if they are specified.
     if (!err.empty()) {
       if (plotOrder != 2) {
         ILOG(Info, Support) << "Non empty graphErrors seen for the plot '" << name
                             << "', which is not a graph, ignoring." << ENDM;
       } else {
-        graphPad->Draw("goff");
-        graphErrors = new TGraphErrors(nEntries, TimeStorage, DataStorage[p], ErrorX[p], ErrorY[p]);
-
-        // We draw on the same plot as the main graph, but only error bars.
-        graphErrors->Draw("SAME E");
-
+        graphErrors->Draw(opt.data());
         // We try to convince ROOT to delete graphErrors together with the rest of the canvas.
-        if (auto* pad = thisCanvas->GetPad(p + 1)) {
+        if (auto* pad = thisCanvas->GetPad(1)) {
           if (auto* primitives = pad->GetListOfPrimitives()) {
-            primitives->Add(graphErrors);
+            primitives->Add(graphErrors); // TO-DO: Is this needed?
           }
         }
       }
     }
-  }
-}
\ No newline at end of file
+  } // Trending vs Slices
+  else if (strcmp(trendType.data(), "slices2D") == 0) {
+
+    thisCanvas->cd(1);
+    const int xBins = axis[0].size();
+    float xBoundaries[xBins];
+    for (int i = 0; i < xBins; i++) {
+      xBoundaries[i] = axis[0][i];
+    }
+    const int yBins = axis[1].size();
+    float yBoundaries[yBins];
+    for (int i = 0; i < yBins; i++) {
+      yBoundaries[i] = axis[1][i];
+    }
+
+    TH2F* graph2D = new TH2F("Graph2D", "", xBins - 1, xBoundaries, yBins - 1, yBoundaries);
```

Comment:
Can several histograms with the same name exist in on e session? If so, root will delete the previous one.

Reply:
Normally this works as the TH2F is will be associated to a primitive of the canvas, but I will check this in more detail for the next bigger update